File: 164-maximum-gap/maximum-gap.py
class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        if len(nums)<2:
            return 0
        maxi=0
        nums=sorted(nums)
        for i in range(0,len(nums)-1):
            maxi=max(nums[i+1]-nums[i],maxi)
        return maxi

File: 164-maximum-gap/maximum-gap.cpp
class Solution {
public:
    int maximumGap(vector<int>& nums) {
        auto n=nums.size();
        if (n<2)
        return 0;
        sort(nums.begin(),nums.end());
        int maxi=0;
        for(auto i=0;i<n-1;i++)
            {
                maxi=max(maxi,nums[i+1]-nums[i]);
            }
        return maxi;
    }
};